File: app/src/main/java/com/navitech/navi/data/model/attractions/Category.java
package com.navitech.navi.data.model.attractions;

import static com.navitech.navi.utils.Constants.KEY_DESCRIPTION;
import static com.navitech.navi.utils.Constants.KEY_NAME;
import static com.navitech.navi.utils.Constants.KEY_TAG;

import com.navitech.navi.data.model.GenericModel;
import com.parse.ParseClassName;
import com.parse.ParseObject;

import java.io.Serializable;

@ParseClassName("category")
public class Category extends GenericModel implements Serializable {
    public Category(){}

    public Category(int tag) {
        setTag(tag);
    }

    public String getName() {
        return getString(KEY_NAME);
    }

    public void setName(String name) {
        put(KEY_NAME, name);
    }

    public String getDescription() {
        return getString(KEY_DESCRIPTION);
    }

    public void setDescription(String name) {
        put(KEY_DESCRIPTION, name);
    }

    public int getTag() {
        return getInt(KEY_TAG);
    }

    public void setTag(int tag) {
        put(KEY_TAG, tag);
    }
}

File: app/src/main/java/com/navitech/navi/data/model/travel/Category.kt
package com.navitech.navi.data.model.travel

import com.navitech.navi.data.model.GenericModel
import com.navitech.navi.utils.Constants.KEY_DESCRIPTION
import com.navitech.navi.utils.Constants.KEY_NAME
import com.navitech.navi.utils.Constants.KEY_TAG
import com.parse.ParseClassName
import java.io.Serializable

@ParseClassName("category")
class Category() : GenericModel(), Serializable {
    var name: String
        get() = getString(KEY_NAME)
        set(name) {
            put(KEY_NAME, name)
        }
    var description: String
        get() = getString(KEY_DESCRIPTION)
        set(name) {
            put(KEY_DESCRIPTION, name)
        }
    var tag: Int
        get() = getInt(KEY_TAG)
        set(tag) {
            put(KEY_TAG, tag)
        }
}